File: python-php-API/mashery.py
import base64
import requests

API_KEY = 'nqzum2aq44amvq8g8b8wbqx4'
SECRET_KEY = '2FgCD9uN8b'
TOKEN_ENDPOINT = 'https://api.manheim.com/oauth2/token.oauth2'

header = ""
token = ""
def get_encoded_header():
    authorization_string = API_KEY + ":" + SECRET_KEY
    authorization_string_bytes = authorization_string.encode("ascii")

    authorization_header_encoded = base64.b64encode(authorization_string_bytes)
    return authorization_header_encoded.decode('ascii')


def get_access_token(authorization_header_encoded):
    r = requests.post(TOKEN_ENDPOINT,
                      headers={'Authorization': 'Basic ' + authorization_header_encoded,
                               'content-type': 'application/x-www-form-urlencoded'},
                      data={'grant_type': 'client_credentials'})
    #token = r.json()['access_token']
    
    return r.json()


def has_token_expires(token):
    r = requests.get('https://api.manheim.com/oauth2/token/status',
                     headers={'Authorization': f'{token["token_type"]} {token["access_token"]}'})
    if r.status_code == 200:
        return False
    else:
        return True


def get_data(token):
    r = requests.get('https://api.manheim.com/valuations/vin/1VWAH7A34DC146014?include=retail,historical,forecast',
                      headers={'Authorization': f'{token["token_type"]} {token["access_token"]}'})
    
    print(r.json())

header = get_encoded_header()
token = get_access_token(header)
get_data(token)
#print(has_token_expires(token))



#token = {'token_type': 'Bearer', 'access_token': 'tdvmtdz436rb6tm7evaajvpw'}
#if has_token_expires(token):
#    print("Token expired. Generating new one")
#    header = get_encoded_header()
#    token = get_access_token(header)

#print("Token", token)
#get_data(token)

File: python-php-API/mashery.php
<?php
   function get_encoded_header(){
       $API_KEY = "nqzum2aq44amvq8g8b8wbqx4";
       $SECRET_KEY = "2FgCD9uN8b";
       $authorization_string = $API_KEY.":".$SECRET_KEY;
       $authorization_header_encoded = base64_encode($authorization_string);
       return  $authorization_header_encoded;
   }

   function get_access_token($authorization_header_encoded){
        $TOKEN_ENDPOINT = 'https://api.manheim.com/oauth2/token.oauth2';
        $data = array('grant_type' => 'client_credentials');
        $options = array(
            'http' => array(
                'header'  => "Authorization: Basic ".$authorization_header_encoded."\r\n", "Content-type: application/x-www-form-urlencoded",
                'method'  => 'POST',
                'content' => http_build_query($data)
            )
        );
        $context  = stream_context_create($options);

        $result = file_get_contents($TOKEN_ENDPOINT, false, $context);
        if ($result === FALSE) { /* Handle error */ }
        return json_decode($result,true);
   }
   
   function has_token_expires($token){
        $opts = array(
            'http'=>array(
            'method'=>"GET",
            'header'=>"Authorization: ".$token['token_type']." ".$token['access_token']
            )
        );
        
        $context = stream_context_create($opts);
        
        // Open the file using the HTTP headers set above
        $file = file_get_contents('https://api.manheim.com/oauth2/token/status', false, $context);

        #$status = json_decode($file,true);
        $status_line = $http_response_header[0];
        preg_match('{HTTP\/\S*\s(\d{3})}', $status_line, $match);
        $status = $match[1];
        
        if ($status == 200)
            return False;
        else
            return True;
   }

   function get_data($token = NULL){
        
        if($token ==NULL || has_token_expires($token)!=True)
            {
                $header = get_encoded_header();
                $token = get_access_token($header);
            }

        $opts = array(
            'http'=>array(
            'method'=>"GET",
            'header'=>"Authorization: ".$token['token_type']." ".$token['access_token']
            )
        );
        
        $context = stream_context_create($opts);
        // Open the file using the HTTP headers set above
        $file = file_get_contents('https://api.manheim.com/valuations/vin/1VWAH7A34DC146014?include=retail,historical,forecast', false, $context);

        return json_decode($file,true);
   }
   $get_data = get_data(NULL);
   var_dump($get_data);
?>